lib/csf-tools/src/ConfigFile.ts

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable no-underscore-dangle */
+import fs from 'fs-extra';
+import * as t from '@babel/types';
+import generate from '@babel/generator';
+import traverse from '@babel/traverse';
+import { babelParse } from './babelParse';
+
+const logger = console;
+
+const _getPath = (path: string[], node: t.Node): t.Node | undefined => {
+  if (path.length === 0) {
+    return node;
+  }
+  if (t.isObjectExpression(node)) {
+    const [first, ...rest] = path;
+    const field = node.properties.find((p: t.ObjectProperty) => {
+      return t.isIdentifier(p.key) && p.key.name === first;
+    });
+    if (field) {
+      return _getPath(rest, (field as t.ObjectProperty).value);
+    }
+  }
+  return undefined;
+};
+
+const _findVarInitialization = (identifier: string, program: t.Program) => {
+  let init: t.Expression = null;
+  let declarations: t.VariableDeclarator[] = null;
+  program.body.find((node: t.Node) => {
+    if (t.isVariableDeclaration(node)) {
+      declarations = node.declarations;
+    } else if (t.isExportNamedDeclaration(node) && t.isVariableDeclaration(node.declaration)) {
+      declarations = node.declaration.declarations;
+    }
+
+    return (
+      declarations &&
+      declarations.find((decl: t.Node) => {
+        if (
+          t.isVariableDeclarator(decl) &&
+          t.isIdentifier(decl.id) &&
+          decl.id.name === identifier
+        ) {
+          init = decl.init;
+          return true; // stop looking
+        }
+        return false;
+      })
+    );
+  });
+  return init;
+};
+
+const _makeObjectExpression = (path: string[], value: t.Expression): t.Expression => {
+  if (path.length === 0) return value;
+  const [first, ...rest] = path;
+  const innerExpression = _makeObjectExpression(rest, value);
+  return t.objectExpression([t.objectProperty(t.identifier(first), innerExpression)]);
+};
+
+const _updateExportNode = (path: string[], expr: t.Expression, existing: t.ObjectExpression) => {
+  const [first, ...rest] = path;
+  const existingField = existing.properties.find((p: t.ObjectProperty) => {
+    return t.isIdentifier(p.key) && p.key.name === first;
+  }) as t.ObjectProperty;
+  if (!existingField) {
+    existing.properties.push(
+      t.objectProperty(t.identifier(first), _makeObjectExpression(rest, expr))
+    );
+  } else if (t.isObjectExpression(existingField.value) && rest.length > 0) {
+    _updateExportNode(rest, expr, existingField.value);
+  } else {
+    existingField.value = _makeObjectExpression(rest, expr);
+  }
+};
+
+export class ConfigFile {
+  _ast: t.File;
+
+  _exports: Record<string, t.Expression> = {};
+
+  _exportsObject: t.ObjectExpression;
```

Comment:
Could you use private fields here?
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields#syntax